R-package/tests/testthat/test_learning_to_rank.R

```diff
@@ -3,6 +3,8 @@ context("Learning to rank")
 # numerical tolerance to use when checking metric values
 TOLERANCE <- 1e-06
 
+ON_SOLARIS <- Sys.info()['sysname'] == "SunOS"
```

Comment:
To fix lint error from CI.
```suggestion
ON_SOLARIS <- Sys.info()["sysname"] == "SunOS"
```